From 30fb1a92b9d0899a2f25cf1917000aad2f4578c5 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 4 Jul 2016 20:14:52 -0700
Subject: [PATCH 4/4] clang: Check for float ABI on arm

Triggering hard-float on arm does not just
depend upong triplet being *-*-eabihf but
also check for float ABI which can be
set on commandline

Signed-off-by: Khem Raj <raj.khem@gmail.com>

xx
---
 lib/Driver/ToolChains.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 706612d..8bac35a 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4169,7 +4169,8 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     default:
       ArchName = Triple.getArchName().str();
     }
-    if (Triple.getEnvironment() == llvm::Triple::MuslEABIHF)
+    if (Triple.getEnvironment() == llvm::Triple::MuslEABIHF ||
+        tools::arm::getARMFloatABI(*this, Args) == tools::arm::FloatABI::Hard)
       ArchName += "hf";
 
     return "/lib/ld-musl-" + ArchName + ".so.1";
-- 
2.9.0

